lib/ansible/modules/cloud/amazon/dynamodb.py

```diff
@@ -0,0 +1,390 @@
+    returned: success
+    type: int
+    sample: 9589
+consumed_capacity:
+    description: the capacity units when you peform a 'get' action
+    returned: success
+    type: dict
+count:
+    description: the number of items in the response when you perform a 'get' action
+    returned: success
+    type: int
+    sample: 13
+response_metadata:
+  description: aws response metadata
+  returned: success
+  type: dict
+  sample:
+    http_headers:
+      content-length: 1490
+      content-type: text/xml
+      date: Tue, 07 Feb 2017 16:43:04 GMT
+      x-amz-crc32: 2745614147
+      x-amzn-requestid: 7f436dea-ed54-11e6-a04c-ab2372a1f14d
+    http_status_code: 200
+    request_id: 7f436dea-ed54-11e6-a04c-ab2372a1f14d
+    retry_attempts: 0
+'''
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, AWSRetry
+
+
+try:
+    import botocore
+except ImportError:
+    pass    # Handled by AnsibleAWSModule
+
+
+@AWSRetry.exponential_backoff(retries=5, delay=5)
+def get_with_backoff(connection, **kwargs):
+    paginator = connection.get_paginator('scan')
+    return paginator.paginate(**kwargs).build_full_result()
+
+
+def get(connection, table, filter_expression, expression_attribute_names,
+        expression_attribute_values):
+
+    if expression_attribute_names:
+        kwargs = {"TableName": table,
+                  "FilterExpression": filter_expression,
+                  "ExpressionAttributeValues": expression_attribute_values,
+                  "ExpressionAttributeNames": expression_attribute_names}
+
+    else:
+        kwargs = {"TableName": table,
+                  "FilterExpression": filter_expression,
+                  "ExpressionAttributeValues": expression_attribute_values}
+
+    response = get_with_backoff(connection, **kwargs)
+
+    changed = False
+
+    response['returned_items'] = response.pop('Items')
+
+    return response, changed
+
+
+def put(connection, table, item):
+
+    response = connection.put_item(
+        TableName=table,
+        Item=item,
+    )
+
+    changed = True
+    return response, changed
+
+
+def update(connection, table, primary_key, update_expression,
+           condition_expression, expression_attribute_names,
+           expression_attribute_values):
+
+    if condition_expression:
+        response = connection.update_item(TableName=table, Key=primary_key, UpdateExpression=update_expression,
+                                          ConditionExpression=condition_expression, ExpressionAttributeValues=expression_attribute_values)
+    elif expression_attribute_names:
+        response = connection.update_item(TableName=table, Key=primary_key, UpdateExpression=update_expression,
+                                          ExpressionAttributeValues=expression_attribute_values, ExpressionAttributeNames=expression_attribute_names)
+    elif update_expression and expression_attribute_values:
```

Comment:
are `update_expression` and `expression_attribute_values` required together - if so that should be expressed in the argument spec